app/scripts/migrations/088.test.ts

```diff
@@ -0,0 +1,638 @@
+import { migrate } from './088';
+
+describe('migration #88', () => {
+  it('updates the version metadata', async () => {
+    const oldStorage = {
+      meta: { version: 0 },
+      data: {},
+    };
+
+    const newStorage = await migrate(oldStorage);
+
+    expect(newStorage.meta).toStrictEqual({ version: 88 });
+  });
+
+  it('returns the state unaltered if it has no NftController property', async () => {
+    const oldData = {
+      some: 'data',
+    };
+    const oldStorage = {
+      meta: { version: 0 },
+      data: oldData,
+    };
+
+    const newStorage = await migrate(oldStorage);
+
+    expect(newStorage.data).toStrictEqual(oldData);
+  });
+
+  it('returns the state unaltered if the NftController object has no allNftContracts property', async () => {
+    const oldData = {
+      NftController: {
+        some: 'data',
+      },
+    };
+    const oldStorage = {
+      meta: { version: 0 },
+      data: oldData,
+    };
+
+    const newStorage = await migrate(oldStorage);
+
+    expect(newStorage.data).toStrictEqual(oldData);
+  });
+
+  it('returns the state unaltered if NftController.allNftContracts is not an object', async () => {
+    const oldData = {
+      NftController: {
+        allNftContracts: 'foo',
+      },
+    };
+    const oldStorage = {
+      meta: { version: 0 },
+      data: oldData,
+    };
+
+    const newStorage = await migrate(oldStorage);
+
+    expect(newStorage.data).toStrictEqual(oldData);
+  });
+
+  it('returns the state unaltered if any value of the NftController.allNftContracts object is not an object itself', async () => {
+    const oldData = {
+      NftController: {
+        allNftContracts: {
+          '0x111': {
+            '123': 'foo',
+          },
+          '0x222': 'bar',
+        },
+      },
+    };
+    const oldStorage = {
+      meta: { version: 0 },
+      data: oldData,
+    };
+
+    const newStorage = await migrate(oldStorage);
+
+    expect(newStorage.data).toStrictEqual(oldData);
+  });
+
+  it('rewrites NftController.allNftContracts so that decimal chain IDs are converted to hex strings', async () => {
+    const oldStorage = {
+      meta: { version: 0 },
+      data: {
+        NftController: {
+          allNftContracts: {
+            '0x111': {
+              '16': ['contract 1', 'contract 2'],
+              '32': ['contract 3', 'contract 4'],
+            },
+            '0x222': {
+              '64': ['contract 5', 'contract 6'],
+              '128': ['contract 7', 'contract 8'],
```

Comment:
I realize the migration doesn't actually touch inside these arrays but wondering if they should be arrays of objects atleast to match the real shape?

Reply:
Sure! Fixed in [d274c4d](https://github.com/MetaMask/metamask-extension/pull/19472/commits/d274c4d6672dcdf0e12c497aff706d3c665e7d2d).